app/models/health_care_application.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+class HealthCareApplication < ActiveRecord::Base
+  validates(:state, presence: true, inclusion: %w(success error failed pending))
+  validates(:form_submission_id_string, :timestamp, presence: true, if: :success?)
+
+  def success?
+    state == 'success'
+  end
+
+  def set_result!(result)
```

Comment:
set_result! seems like a weird name for the method to call on success, so I was confused about why it's always setting the state to 'success'. Even `set_result_on_success!` would be clearer.